### PR TITLE
v4 fixes for 5702 and 5714

### PR DESCRIPTION
**Pull request by kflynn**
This is @sekar-saravanan's fixes for #5702 and #5714 combined with my test cases.

Reviewers, this is **definitely** best reviewed commit by commit.

Note also that **irtype.go is known to be incomplete** -- but, as demonstrated, it's already good enough to do useful testing work!

Finally, note also that I have _not_ tried to backport this work to the v3 branch. It's definitely doable, but since this involves a couple of chainsaw commits, there'll be some weirdness around the Go import paths, I'm sure.

### CODE REVIEW

**Comment by kflynn** [comment]
(I initially submitted this as a PR against `master`, which is why the labels were so disturbing at first. 🙂)

**Comment by kflynn** [comment]
@sekar-saravanan and @the-wondersmith, it'd be good for y'all to look at this too.

File: cmd/entrypoint/irtype.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+type IRClusterTarget struct {
+	IP         string `json:"ip"`
+	Port       int    `json:"port"`
+	TargetKind string `json:"target_kind"`
+}
+
+type IRCluster struct {
+	IRResource
+	BarHostname      string               `json:"_hostname"`  // Why this _and_ hostname?
+	BarNamespace     string               `json:"_namespace"` // Why this _and_ namespace?
```
Comment (AliceProxy) [inline cmd/entrypoint/irtype.go:33]: I don't like this but I get why it's here for now

**Review by AliceProxy — APPROVED** [review]
seems fine to me. I'm not going to bother nit picking until CI is in a place where tests are passing and reliable indicators again and when the PR for merging this branch into main comes.

File: cmd/entrypoint/irtype.go (hunk 1 above)
Comment (kflynn) [reply cmd/entrypoint/irtype.go:33]: You and me both. I'll figure out what up with this soon.

**Pull request merged**